PeriodogramAnalysis/PeriodogramAnalysis.py

```diff
@@ -1,282 +1,345 @@
 import os
 
+import pathlib
 import numpy as np
 import matplotlib.pyplot as plt
 from scipy import signal as sig
 from scipy.integrate import simpson
 from dataclasses import dataclass, field
-from typing import Optional, Union
+from collections import defaultdict
+from typing import Tuple, Dict, Any, Generator, Optional
 
 from miv.core.operator.operator import OperatorMixin
 from miv.core.operator.wrapper import cache_call
+from miv.core.datatype import Signal
 from miv.typing import SignalType
-from PowerSpectrumAnalysis import SpectrumAnalysis, multitaper_psd
+from MultitaperPowerSpectrum import multitaper_psd
 
-@ dataclass()
+DictType = Dict[int, Dict[int, Dict[str, Any]]]
```

Comment:
Don't do this. It is impossible to see that `DictType` as nested dictionary, and can be misleading

Reply:
ok, I have directly used Dict[int, Dict[int, Dict[str, Any]]], previously I thought it is too long.